schema/manifestlist_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/opencontainers/image-spec/schema"
+)
+
+func TestManifestList(t *testing.T) {
+	for i, tt := range []struct {
+		manifestList string
+		fail         bool
+	}{
+		// expected failure: mediaType does not match pattern
+		{
+			manifestList: `
+{
+  "schemaVersion": 2,
+  "mediaType": "invalid"
```

Comment:
Probably also want a test where `schemaVersion` is 0, although support for checking that is still in flight with #404.